src/webhook.rs

```diff
@@ -732,11 +688,151 @@ async fn handle_comment(
+		let member = github_bot
+			.org_member(&owner, &requested_by)
+			.await
+			.map_err(|e| {
+				Error::OrganizationMembership {
+					source: Box::new(e),
+				}
+				.map_issue(Some((
+					owner.to_string(),
+					repo_name.to_string(),
+					number,
+				)))
+			})?;
+		if member != 204 {
+			Err(Error::OrganizationMembership {
+				source: Box::new(Error::Message {
+					msg: format!(
+						"{} is not a member of {}; aborting.",
+						requested_by, owner
+					),
+				}),
+			}
+			.map_issue(Some((
+				owner.to_string(),
+				repo_name.to_string(),
+				number,
+			))))?;
+		}
+
+		handle_burnin_request(
+			github_bot,
+			&state.gitlab_bot,
+			&state.matrix_bot,
+			owner,
+			&requested_by,
+			&repo_name,
+			&pr,
+		)
+		.await;
 	}
 
 	Ok(())
 }
 
+async fn handle_burnin_request(
+	github_bot: &GithubBot,
+	gitlab_bot: &GitlabBot,
+	matrix_bot: &MatrixBot,
+	owner: &str,
+	requested_by: &str,
+	repo_name: &str,
+	pr: &PullRequest,
+) {
+	let make_job_link =
+		|url| format!("<a href=\"{}\">CI job for burn-in deployment</a>", url);
+
+	let unexpected_error_msg = "Starting CI job for burn-in deployment failed with an unexpected error; see logs.".to_string();
+	let mut matrix_msg = String::from("");
+
+	let msg = match gitlab_bot.build_artifact(&pr.head.sha) {
+		Ok(job) => {
+			let ci_job_link = make_job_link(job.url);
+
+			match job.status {
+				JobStatus::Started => {
+					format!("{} was started successfully.", ci_job_link)
+				}
+				JobStatus::AlreadyRunning => {
+					format!("{} is already running.", ci_job_link)
+				}
+				JobStatus::Finished => format!(
+					"{} already ran and finished with status `{}`.",
+					ci_job_link, job.status_raw,
+				),
+				JobStatus::Unknown => format!(
+					"{} has unexpected status `{}`.",
+					ci_job_link, job.status_raw,
+				),
+			}
+		}
+		Err(e) => {
+			log::error!("handle_burnin_request: {}", e);
+			match e {
+				Error::GitlabJobNotFound { commit_sha } => format!(
+					"No matching CI job was found for commit `{}`",
+					commit_sha
+				),
+				Error::StartingGitlabJobFailed { url, status, body } => {
+					let ci_job_link = make_job_link(url);
+
+					matrix_msg = format!(
+						"Starting {} failed with HTTP status {} and body: {}",
+						ci_job_link, status, body,
+					);
+
+					format!(
+						"Starting {} failed with HTTP status {}",
+						ci_job_link, status,
+					)
+				}
+				Error::GitlabApi {
+					method,
+					url,
+					status,
+					body,
+				} => {
+					matrix_msg = format!(
+						"Request {} {} failed with reponse status {} and body: {}",
+						method, url, status, body,
+					);
+
+					unexpected_error_msg
+				}
+				_ => unexpected_error_msg,
+			}
+		}
+	};
+
+	let _ = github_bot
+		.create_issue_comment(owner, &repo_name, pr.number, &msg)
+		.await
+		.map_err(|e| {
+			log::error!("Error posting GitHub comment: {}", e);
+		});
+
+	let m = if matrix_msg.len() > 0 {
+		matrix_msg
+	} else {
+		msg
+	};
```

Comment:
If `matrix_msg` is an `Option` you can do:

```suggestion
  let m = matrix_msg.unwrap_or(msg);
```

Reply:
Ahh, this is the kind of spoonfeeding of idiomatic language use I came here for. :)